### PR TITLE
Opt remove async op

### PR DESCRIPTION
**Pull request by bensander**

### CODE REVIEW

**Comment by whchung** [comment]
@bensander could you summarize the effect of the PR? From the look of it it's a performance improvement but I'd like to hear more from you

**Comment by bensander** [comment]
Yes perf improvement.   Around 7-8% on important workloads , in particular cases which submit large numbers of commands into the queue before synchronizing.

On Oct 7, 2017, at 1:19 PM, Wen-Heng (Jack) Chung <notifications@github.com<mailto:notifications@github.com>> wrote:


@bensander<https://github.com/bensander> could you summarize the effect of the PR? From the look of it it's a performance improvement but I'd like to hear more from you

**Review by whchung — APPROVED** [review]
LGTM
